flytekit/bin/entrypoint.py

```diff
@@ -96,7 +96,11 @@ def _dispatch_execute(
             c: OR if an unhandled exception is retrieved - record it as an errors.pb
     """
     output_file_dict = {}
-    try:
+
+    @_scopes.system_entry_point
```

Comment:
Although `_dispatch_execute` is already being executed in a system exception scope, that scope is too far up the stack to wrap exceptions raised in this function so the subsequent except blocks below can handle them.